src/common/components/recovery-account/index.tsx

```diff
@@ -65,12 +65,12 @@ export default function AccountRecovery(props: Props) {
     const account = await getAccount(props.activeUser!.username);
     setAccountData(account);
     const { recovery_account } = account;
-    setCurrRecoveryAccount(recovery_account);
+    setCurrRecoveryAccount(ECENCY);
     if (recovery_account === props.activeUser?.username) {
       setToWarning(_t("account-recovery.same-recover-agent-suggestion"));
     }
 
-    if (recovery_account === ECENCY) {
+    if (ECENCY === ECENCY) {
```

Comment:
@MTSaeed this condition always true, why needed?

Reply:
@feruzm My mistake😬
I set this condition to test for the case of ECENCY and after that I forgot to reset that condition. Now i update the condition.
Thanks!

Reply:
@MTSaeed I just tried, it is sending email to backend but not changing recovery agent. Also when I type into new agent field, it sometimes says account doesn't exist error. You should make sure email update only happens if I set new agent to current agent, that way, it should only change email without blockchain broadcast, otherwise, it should broadcast to blockchain as well as send email change to backend with keys.

Reply:
Also please pull before fix above issue, as I pushed couple minor improvements.

Reply:
Hi @feruzm 
I have done some changes in the Logic. I am going to explain the use cases. If you feel to change in any use case, please mention the use case number too.

1- If the current Recovery agent is any account other than ECENCY and you want to set new recover agent other than ECENCY, it would perform the broadcast transaction.

2- If the current Recovery agent is any account other than ECENCY and you want to set ECENCY as your new recover agent, it would send API call to add email in the database and as well as it will perform broadcast transaction to change recover agent to ECENCY.

3- If the current recovery agent is ECENCY and you just want to update the email, it will allow you to update the email and it will not show you the PopOver and also it will not perform broadcast transaction. it will just update the email by sending API call to the Database.

4- If the current recovery agent is ECENCY and user want to set any account other than ECENCY to their new recover agent, it will perform broadcast transaction to update the new recover agent and also send the API call to delete the recent email.

Reply:
@feruzm This is just the video of use case 1 & 2 when the current recover agent is other than ecency.

https://user-images.githubusercontent.com/106739598/232410999-e3d26570-8af7-4122-99ae-20d0aec0c4a7.mp4

Reply:
@MTSaeed 4. Do not delete email from backend, allow user to change recovery agent broadcasting transaction, but no api call needed.

Reply:
> @MTSaeed 4. Do not delete email from backend, allow user to change recovery agent broadcasting transaction, but no api call needed.

Updated. Now if user try to change recover agent from ECENCY to any other account, it will not send API call to delete the email from database. It will just broadcast the transaction.

Reply:
@MTSaeed consider using Debounce for account typing, so we don't do multiple calls to rpc server unnecessarily. debounce 500 ms should be sufficient.

Reply:
yes. updated.